cfg_support/src/lib.rs

```diff
@@ -0,0 +1,335 @@
+    vendor: Option<&'a str>,
+}
+
+#[derive(Copy, Clone)]
+pub enum CfgBuilder<'a> {
+    Uninitialized,
+    Cond(Cfg<'a>),
+}
+
+fn check_same(target: String, other: Option<&'_ str>) -> bool {
+    Some(target) == other.map(|x: &str| x.to_uppercase())
+}
+fn check_conflict(target: String, other: Option<&'_ str>) -> bool {
+    None == other || check_same(target, other)
+}
+
+impl<'a> Cfg<'a> {
+    pub fn arch_conflict(&'a self, arg: &CfgTarget) -> bool {
+        check_conflict((&*arg.target_arch).into(), self.arch)
+    }
+
+    pub fn os_conflict(&'a self, arg: &CfgTarget) -> bool {
+        check_conflict((&*arg.target_os).into(), self.os)
+    }
+
+    pub fn family_conflict(&'a self, arg: &CfgTarget) -> bool {
+        check_conflict((&*arg.target_family).into(), self.family)
+    }
+
+    pub fn env_conflict(&'a self, arg: &CfgTarget) -> bool {
+        check_conflict((&*arg.target_env).into(), self.env)
+    }
+
+    pub fn endian_conflict(&'a self, arg: &CfgTarget) -> bool {
+        check_conflict((&*arg.target_endian).into(), self.endian)
+    }
+
+    pub fn pointer_width_conflict(&'a self, arg: &CfgTarget) -> bool {
+        check_conflict((&*arg.target_pointer_width).into(), self.pointer_width)
+    }
+
+    pub fn vendor_conflict(&'a self, arg: &CfgTarget) -> bool {
+        check_conflict((&*arg.target_vendor).into(), self.vendor)
+    }
+
+    pub fn check(&'a self, arg: &CfgTarget) -> bool {
+        self.arch_conflict(arg)
+            && self.os_conflict(arg)
+            && self.family_conflict(arg)
+            && self.env_conflict(arg)
+            && self.endian_conflict(arg)
+            && self.pointer_width_conflict(arg)
+            && self.vendor_conflict(arg)
+    }
+}
+
+impl<'a> CfgTarget {
+    pub fn new() -> CfgTarget {
+        fn getenv(var: &'static str) -> String {
+            std::env::var(var)
+                .unwrap_or_else(|_| String::new())
+                .to_uppercase()
+        }
+        CfgTarget {
+            target_arch: getenv("CARGO_CFG_TARGET_ARCH"),
+            target_feature: getenv("CARGO_CFG_TARGET_FEATURE"),
+            target_os: getenv("CARGO_CFG_TARGET_OS"),
+            target_family: getenv("CARGO_CFG_TARGET_FAMILY"),
+            target_env: getenv("CARGO_CFG_TARGET_ENV"),
+            target_endian: getenv("CARGO_CFG_TARGET_ENDIAN"),
+            target_pointer_width: getenv("CARGO_CFG_TARGET_POINTER_WIDTH"),
+            target_vendor: getenv("CARGO_CFG_TARGET_VENDOR"),
+        }
+    }
+    // Same and conflict should differ by the None case.
+
+    pub fn any(&self, args: &'a [Cfg<'a>]) -> bool {
+        assert_eq!(args.is_empty(), false);
+        args.iter().any(|other| other.check(self))
+    }
+
+    pub fn all(&self, args: &'a [Cfg<'a>]) -> bool {
+        assert_eq!(args.is_empty(), false);
+        args.iter().all(|other| other.check(self))
+    }
+}
+
+impl Default for CfgTarget {
+    fn default() -> Self {
+        Self::new()
+    }
+}
```

Comment:
When this construct comes up, it means that the `new()` function should just be removed and turned into the implementation of the `default()` function.

Reply:
The only issue I think is that default() when gobbling up the environment variables, is non-deterministic, and there is some discussion on whether that is an appropriate default.

Anyhow, I did as you mentioned deterministic or not it's the only way to construct one!